packages/ra-core/src/sideEffect/redirection.js

```diff
@@ -30,7 +31,7 @@ export function* handleRedirection({
                   )
               )
           )
-        : yield put(reset('record-form')); // explicit no redirection, reset the form
+        : yield put(reset(formName)); // explicit no redirection, reset the form
```

Comment:
If I understand correctly, this side effect is the reason why you pass the form name to all the data actions? Because I've been wondering why you updated all the action creators.